wallet/src/de/schildbach/wallet/ui/widget/ShortcutsPane.kt

```diff
@@ -100,42 +111,54 @@ class ShortcutsPane(context: Context, attrs: AttributeSet) : LinearLayout(contex
                 refresh()
                 return false
             }
-
         }
         viewTreeObserver.addOnPreDrawListener(onPreDrawListener)
     }
 
     fun setup() {
         addShortcut(secureNowButton)
-        secondaryItems.add(scanToPayButton)
-        if (isSmallScreen) {
-            secondaryItems.add(receiveButton)
+        if (userHasBalance) {
+            secondaryItems.add(scanToPayButton)
+        }
+        secondaryItems.add(receiveButton)
+
+        if (userHasBalance) {
+            secondaryItems.add(payToAddressButton)
         }
-        secondaryItems.add(payToAddressButton)
         secondaryItems.add(buySellButton)
         secondaryItems.forEach {
             addShortcut(it)
         }
-        if (!isSmallScreen) {
-            addShortcut(receiveButton)
-        }
+
+        addShortcut(explore)
     }
 
     private fun refresh() {
         val displayed = mutableSetOf<ShortcutButton>()
-        secureNowButton.visibility = if (showSecureNow) {
+
+        secureNowButton.isVisible = showSecureNow
+        if (secureNowButton.isVisible) {
             displayed.add(secureNowButton)
-            View.VISIBLE
-        } else View.GONE
-        if (!isSmallScreen) {
-            displayed.add(receiveButton)
         }
-        val numberOfButtons = if (isSmallScreen) 3 else 4
-        secondaryItems.forEach {
-            it.visibility = if (displayed.size < numberOfButtons) {
-                displayed.add(it)
-                View.VISIBLE
-            } else View.GONE
```

Comment:
For smaller screens, only 3 buttons should be displayed.  The story does not define what should be displayed if there are only three places for shortcut buttons.